### PR TITLE
Action for running gradle command that needs an emulator booted

### PR DESCRIPTION
**Pull request by joshrlesch**
New action to run instrumented tests for android. This basically creates and boots an emulator before running an gradle commands so that you can run instrumented tests against that emulator. After the gradle command is executed, the avd gets shut down and deleted. This is really helpful on CI services, keeping them clean and always having a fresh avd for testing.

This is in reference to the issue I opened here https://github.com/fastlane/fastlane/issues/4190

### CODE REVIEW

**Comment by joshrlesch** [comment]
Ok, not sure why rubocop is failing the tests. I "think" the failure is saying that the `self.run` method is doing to much stuff? I moved the kernal loop outside of it, but still throwing the same error as last time.

Any idea's on this?

File: fastlane/docs/Actions.md [hunk 1]
```diff
@@ -1984,6 +1984,18 @@ adb_devices.each  do | device |
 end
 ```
 
+### instrumented_tests
+
+Creates and boots avd device before running gradle command.
+
+```ruby
+instrumented_tests(
+    avd_name: "Nexus_10",
```
Comment (KrauseFx) [inline fastlane/docs/Actions.md]: Make sure to use 2 spaces here

**Comment by ohwutup** [comment]
Hey @joshrlesch we just pushed out a new plugin infrastructure https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md#create-your-own-plugin. We would really love it if you could reimplement this as a plugin and then you can share it via your own github repo! 🔌 

You should be able to just run `fastlane new_plugin` and then copy your action's logic right over!

**Comment by lexxdark** [comment]
I've created the plugin from your code. If you wish to do so yourself, please ping back and I will take it down, or pass ownership. Have fun. ;)

**Comment by joshrlesch** [comment]
Hey @lexxdark I was hoping to get a plugin for this. I made some improvementsort to this locally and currently testing it and once good will create a plug-in for it.

**Comment by lexxdark** [comment]
I was wondering about the issues that I've encountered. :)
I'm basically doing the same thing, shall we create a project & share?
Otherwise we can each finish the plugin for our own target(s) and sync back later.

Btw when you want me to take down any of the above projects ping me back,

**Comment by joshrlesch** [comment]
Sorry I was away on vacation and busy at work, currently using it in a custom action in house, but still wanted to open a plugin for this with some improvements. Some timeout protection so its not an infinite loop.

**Comment by lexxdark** [comment]
Not an issue, it's kinda normal to transfer ownership of projects in the opensource world. :)
Shall I transfer it to your account? BTW It might clash with your current repo. 

If I have time I will add the timeout tomorow.

**Comment by joshrlesch** [comment]
@lexxdark Cool thanks man I appreciate it, I deleted my repo fyi so should be good.

**Comment by lexxdark** [comment]
Done, we need to do the same on the rubygems website. Do you have an account there?
I'm sad that I did most of the changes (or eqquivalent) done in the other repo :(

**Comment by joshrlesch** [comment]
Thanks for doing that man! I do have a rubygems account, https://rubygems.org/profiles/joshrl not sure what you all need for that.

**Comment by lexxdark** [comment]
Done ;)

**Comment by joshrlesch** [comment]
You are the man, glad we got together on this! And what did mean by sad with most of the changes done in other repo, did you loose any work?

**Comment by lexxdark** [comment]
It's sad when the same work is done twice by two persons. :( 
As I understood FisherKK had done more or less the same changes I did. Possibly better since I'm not an expert in Ruby.

**Comment by joshrlesch** [comment]
Ah ic what your are saying. Yeah not sure if he knew where was a plugin already published by you. But I did mention to him to open a PR with his changes. I hope you still contribute with any improvements you can think of. Again thanks for everything with this, 👍

**Pull request closed without merging**